internal/ingress/controller/store/store_test.go

```diff
@@ -605,37 +619,89 @@ func TestStore(t *testing.T) {
 	// check invalid secret (missing ca)
 }
 
-func createNamespace(clientSet *kubernetes.Clientset, t *testing.T) string {
+func createNamespace(clientSet *fake.Clientset, t *testing.T) string {
 	t.Log("creating temporal namespace")
-	ns, err := framework.CreateKubeNamespace("store-test", clientSet)
+
+	namespace := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "store-test",
+		},
+	}
+
+	ns, err := clientSet.CoreV1().Namespaces().Create(namespace)
 	if err != nil {
-		t.Errorf("unexpected error creating ingress client: %v", err)
+		t.Errorf("unexpected error creating the namespace: %v", err)
 	}
 	t.Logf("temporal namespace %v created", ns)
 
-	return ns
+	return ns.Name
 }
 
-func deleteNamespace(ns string, clientSet *kubernetes.Clientset, t *testing.T) {
-	t.Logf("deleting temporal namespace %v created", ns)
-	err := framework.DeleteKubeNamespace(clientSet, ns)
+func deleteNamespace(ns string, clientSet *fake.Clientset, t *testing.T) {
+	t.Logf("deleting temporal namespace %v", ns)
+
+	err := clientSet.CoreV1().Namespaces().Delete(ns, &metav1.DeleteOptions{})
 	if err != nil {
-		t.Errorf("unexpected error creating ingress client: %v", err)
+		t.Errorf("unexpected error deleting the namespace: %v", err)
 	}
 	t.Logf("temporal namespace %v deleted", ns)
 }
 
-func ensureIngress(ingress *extensions.Ingress, clientSet *kubernetes.Clientset) (*extensions.Ingress, error) {
-	s, err := clientSet.ExtensionsV1beta1().Ingresses(ingress.Namespace).Update(ingress)
+func createConfigMap(clientSet *fake.Clientset, ns string, t *testing.T) string {
+	t.Log("creating temporal config map")
+
+	configMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:     "config",
+			SelfLink: fmt.Sprintf("/api/v1/namespaces/%s/configmaps/config", ns),
+		},
+	}
+
+	cm, err := clientSet.CoreV1().ConfigMaps(ns).Create(configMap)
+	if err != nil {
+		t.Errorf("unexpected error creating the configuration map: %v", err)
+	}
+	t.Logf("required configmap %v created", cm)
+
+	return cm.Name
+}
+
+func deleteConfigMap(cm, ns string, clientSet *fake.Clientset, t *testing.T) {
+	t.Logf("deleting temporal configmap %v", cm)
+
+	err := clientSet.CoreV1().ConfigMaps(ns).Delete(cm, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Errorf("unexpected error deleting the configmap: %v", err)
+	}
+	t.Logf("temporal configmap %v deleted", cm)
+}
+
+func ensureIngress(ingress *extensions.Ingress, clientSet *fake.Clientset, t *testing.T) (*extensions.Ingress, error) {
+	s, err := clientSet.Extensions().Ingresses(ingress.Namespace).Update(ingress)
+
 	if err != nil {
 		if k8sErrors.IsNotFound(err) {
-			return clientSet.ExtensionsV1beta1().Ingresses(ingress.Namespace).Create(ingress)
+			t.Logf("ingress %v not found, creating", ingress)
+			return clientSet.Extensions().Ingresses(ingress.Namespace).Create(ingress)
 		}
 		return nil, err
```

Comment:
You can skip returning errors in tests and simply call `t.Fatal` in case an error occurs, like you did above. Call `t.Helper()` at the beginning of the function in that case.

Reply:
Done